test/spec/controllers/inventoryCtrlSpec.js

```diff
@@ -53,14 +56,36 @@ describe('Inventory Controller', function() {
     expect(scope.selectedPotion.key).to.eql('Base');
   });
 
-  it('hatches a pet', function(){
-    scope.chooseEgg('Cactus');
-    scope.choosePotion('Base');
-    expect(user.items.eggs).to.eql({Cactus: 0});
-    expect(user.items.hatchingPotions).to.eql({Base: 0});
-    expect(user.items.pets).to.eql({'Cactus-Base': 5});
-    expect(scope.selectedEgg).to.eql(null);
-    expect(scope.selectedPotion).to.eql(null);
+  describe('Hatching Pets', function(){
+    beforeEach(function() {
+      sandbox.stub(rootScope, 'openModal');
+    });
+
+    it('hatches a pet', function(){
+      scope.chooseEgg('Cactus');
+      scope.choosePotion('Base');
+      expect(user.items.eggs).to.eql({Cactus: 0});
+      expect(user.items.hatchingPotions).to.eql({Base: 0});
+      expect(user.items.pets).to.eql({'Cactus-Base': 5});
+      expect(scope.selectedEgg).to.eql(null);
+      expect(scope.selectedPotion).to.eql(null);
+    });
+
+    it('shows a modal for pet hatching', function(){
+      scope.chooseEgg('Cactus');
+      scope.choosePotion('Base');
+
+      expect(rootScope.openModal).to.have.been.calledOnce;
+      expect(rootScope.openModal).to.have.been.calledWith('hatchPet');
+    });
+    
+    it('does not show pet hatching modal if user has opted out', function(){
+      user.preferences.suppressModals.hatchPet = true;
+      scope.chooseEgg('Cactus');
+      scope.choosePotion('Base');
+
+      expect(rootScope.openModal).to.not.be.called;
```

Comment:
Needs more of them, though. See my comment above about trouble with `$watch` on some of it.